app/models/trends/statuses.rb

```diff
@@ -96,27 +119,17 @@ def calculate_scores(statuses, at_time)
 
       decaying_score = score * (0.5**((at_time.to_f - status.created_at.to_f) / options[:score_halflife].to_f))
 
-      next unless decaying_score >= options[:decay_threshold]
-
-      global_items << { score: decaying_score, item: status }
-      locale_items[status.language] << { account_id: status.account_id, score: decaying_score, item: status } if valid_locale?(status.language)
+      [decaying_score, status]
     end
 
-    replace_items('', global_items)
-
-    Trends.available_locales.each do |locale|
-      replace_items(":#{locale}", locale_items[locale])
+    Trends::Status.transaction do
+      Trends::Status.upsert_all(items.map { |(score, status)| { id: status.id, account_id: status.account_id, score: score, language: status.language, allowed: status.trendable? } }, unique_by: :id)
+      Trends::Status.where('score < ?', options[:decay_threshold]).delete_all
```

Comment:
What happens if a previously trending status gets deleted? Seems to me the `trends_status` record would remain forever and pollute the table forever, with its score not decaying because there is no associated status to compute from.

This also does not seem to place any limit to the number of items in this table.